### DISCUSSION TITLE
Questions about world info and generation length

### DISCUSSION THREAD

**Discussion by FlyToha** [LostRuins/koboldcpp#838] [Q&A]
Hi! Thanks koboldcpp's developers for this amazing tool, i rly like interact with it. I have some basic questions:
1. How works exactly world info field? How the most correct way/ways to use it? How long text i can use in this field? How many key words i can use there? On wiki page info about this aspects is too short and non clean for me.
2. I working with llm mixtral-8x7b-instruct-v0.1. First time it works very nice and this model got me long replies (5-7k tokens) when i asked about it. Some times the model even told me herself that she had reached the limit of one message and could continue the next part in a new response. But lately I have been unable to get a long answer from her. Despite the fact that the generation length is set to 8k in the settings, it gives me no more than ~ 600 words in any answer. Even if I directly ask you to write an answer of 1000/2000/3000 words, she still gives an answer of 500-600 words. Also i've tried to decrease generation length to lower values (1024/2048/4096), any way i've got stability 500-600 word in answer from the model. What could be the problem?
Mostly i using Koboldcpp 1.62, but i also tried the latest versions 1.63, 1.64.1. No any changes in my case.

**Comment by LostRuins** [comment]
1. World info adds text into the context whenever the *Key* is detected in the story. So it's good for adding conditional data, whereas memory is for adding permanent static data. There is no limit to the amount of world info you can use, you can add as many keys as you like, but world info will consume context space when triggered.

2. Is the answer finished or is the AI still talking? If the AI is still talking, simply press "Generate More" and it will generate additional text. Also, double check your max context size, as well as the max generation size in settings.

**Reply by FlyToha** [reply]
Thank you for the reply.
1. Can you please clean one more aspect: if one of the few key was triggered - all world info will be moved in context space or only that part, which contain data about the triggered key? For example, in my w info placed a few short descriptions of characters (A, B, C, D). So if i asked bot about writing me some episode with char A, then he (bot) will take to context all data form w info about A, B, C, D chars or about A only?

2. The answer is finished. For example, i wrote some short episode's outline and i asked bot to write me more extended and detailed version in 3200 words. He writing me full completed episode with clean the end, but in 500-600 words. It's the finished answer exactly. And of course i've triple checked context size (30k), and generation size in 8k. But! Now it's works well again and bot's answer length is matches with requested values... I can't understand why... Just suppose that this is depending by session length

**Reply by LostRuins** [reply]
Only the key that was triggered will be inserted, the entire value of that key will be input into the context
